src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs

```diff
@@ -32,7 +32,24 @@ internal static DacDeployOptions CreateSchemaCompareOptions(DeploymentOptions de
                 var prop = dacOptions.GetType().GetProperty(deployOptionsProp.Name);
                 if (prop != null)
                 {
-                    prop.SetValue(dacOptions, deployOptionsProp.GetValue(deploymentOptions));
+                    var optionVal = deployOptionsProp.GetValue(deploymentOptions);
+                    if (deployOptionsProp.Name == "DoNotDropObjectTypes" || deployOptionsProp.Name == "ExcludeObjectTypes")
```

Comment:
Could you not check if the type of the value is DeploymentOptionProps? That would make this future proof so you don't have to keep coming back and adding more props to the list here and elsewhere in this code if/when more are added you want to exclude.